### PR TITLE
#49 #55 posts dev

### PR DESCRIPTION
**Pull request by Joshua-Castillo**
_note: Might have to remove client/.eslintcache when trying out, forgot to remove it when pushing_

**Features added:** 
- single post view 
- all posts view
- routing to all posts view from navbar

**- Things to review:** 
- <Redirect /> from post creation is functioning properly,  although post/post_id works fine 
- Bugs when logging in, first attempt displays error, second attempt works fine

### CODE REVIEW

File: client/src/components/pages/CreatePost.jsx [hunk 1]
```diff
@@ -26,29 +27,29 @@ const CreatePost = ({ isAuthenticated, createPost, uploadPicture, deletePicture,
...
+        if (res.post) {
+            console.log(res.post._id);
+            setPostID(res.post._id);
         }
-    }
+    };
+
+    if (postID) {
+        return <Redirect to={`/post/${postID}`} />
+    };
```
Comment (gaticosalas) [inline client/src/components/pages/CreatePost.jsx:48]: There seemed to be an issue with calling redirect inside the async submit function, so I just saved the post_id to the state and called the redirect in an if statement instead. Works fine now!

File: client/src/components/pages/Post.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { Fragment, useEffect } from 'react'
+import { useParams } from 'react-router-dom';
+import { connect } from 'react-redux'
+import { fetchPost } from '../../actions/posts'
+import Moment from 'react-moment'
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+const Post = ({ isAuthenticated, post: { username, picture, caption, avatar, likes, date }, fetchPost, isFetching }) => {
```
Comment (gaticosalas) [inline client/src/components/pages/Post.jsx:8]: I like the this destructuring directly in the attributes!

**Review by gaticosalas — APPROVED** [review]
Amazing job josh, this was really good!

**Review by vishybala — APPROVED** [review]
very well done :)

**Pull request merged**